ldap/servers/plugins/memberof/memberof.c

```diff
@@ -745,115 +745,93 @@ memberof_call_foreach_dn(Slapi_PBlock *pb __attribute__((unused)), Slapi_DN *sdn
-        if (!all_backends) {
-            be = slapi_be_select(sdn);
-            if (be == NULL) {
-                break;
-            }
-        }
-        if ((base_sdn = (Slapi_DN *)slapi_be_getsuffix(be, 0)) == NULL) {
+        search_pb = slapi_pblock_new();
+        be = slapi_get_first_backend(&cookie);
+        while (be) {
+            PRBool do_suffix_search = PR_TRUE;
             if (!all_backends) {
-                break;
-            } else {
-                /* its ok, goto the next backend */
-                be = slapi_get_next_backend(cookie);
-                continue;
+                be = slapi_be_select(sdn);
+                if (be == NULL) {
+                    break;
+                }
+            }
+            if ((base_sdn = (Slapi_DN *) slapi_be_getsuffix(be, 0)) == NULL) {
+                if (!all_backends) {
+                    break;
+                } else {
+                    /* its ok, goto the next backend */
+                    be = slapi_get_next_backend(cookie);
+                    continue;
+                }
             }
-        }
 
-        if (config->entryScopes || config->entryScopeExcludeSubtrees) {
-            if (memberof_entry_in_scope(config, base_sdn)) {
-                /* do nothing, entry scope is spanning
-                 * multiple suffixes, start at suffix */
-            } else if (config->entryScopes) {
-                for (size_t i = 0; config->entryScopes[i]; i++) {
-                    if (slapi_sdn_issuffix(config->entryScopes[i], base_sdn)) {
-                        /* Search each include scope */
-                        slapi_search_internal_set_pb(search_pb, slapi_sdn_get_dn(config->entryScopes[i]),
-                                                     LDAP_SCOPE_SUBTREE, filter_str, 0, 0, 0, 0,
-                                                     memberof_get_plugin_id(), 0);
-                        slapi_search_internal_callback_pb(search_pb, callback_data, 0, callback, 0);
-                        /* We already did the search for this backend, don't
-                         * do it again when we fall through */
-                        do_suffix_search = PR_FALSE;
+            if (config->entryScopes || config->entryScopeExcludeSubtrees) {
+                if (memberof_entry_in_scope(config, base_sdn)) {
+                    /* do nothing, entry scope is spanning
+                     * multiple suffixes, start at suffix */
+                } else if (config->entryScopes) {
+                    for (size_t i = 0; config->entryScopes[i]; i++) {
+                        if (slapi_sdn_issuffix(config->entryScopes[i], base_sdn)) {
+                            /* Search each include scope */
+                            slapi_search_internal_set_pb(search_pb, slapi_sdn_get_dn(config->entryScopes[i]),
+                                    LDAP_SCOPE_SUBTREE, filter_str, 0, 0, 0, 0,
+                                    memberof_get_plugin_id(), 0);
```

Comment:
Just to keep it consistent with the rest of the code, could you, please, fix the indentation here:

```
                            slapi_search_internal_set_pb(search_pb, slapi_sdn_get_dn(config->entryScopes[i]),
                                                         LDAP_SCOPE_SUBTREE, filter_str, 0, 0, 0, 0,
                                                         memberof_get_plugin_id(), 0);
```